app/uk/gov/hmrc/merchandiseinbaggage/service/CustomsDutyCalculator.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ */
+
+package uk.gov.hmrc.merchandiseinbaggage.service
+
+import java.time.LocalDate
+
+import cats.data.EitherT
+import cats.instances.future._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import uk.gov.hmrc.merchandiseinbaggage.connectors.CurrencyConversionConnector
+import uk.gov.hmrc.merchandiseinbaggage.model.api.{CalculationRequest, CurrencyConversionResponse}
+import uk.gov.hmrc.merchandiseinbaggage.model.core.AmountInPence._
+import uk.gov.hmrc.merchandiseinbaggage.model.core.{AmountInPence, BusinessError, CurrencyNotFound}
+import uk.gov.hmrc.merchandiseinbaggage.repositories.CustomsRate._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+trait CustomsDutyCalculator extends CurrencyConversionConnector {
+
+  def customDuty(httpClient: HttpClient, calculationRequest: CalculationRequest)
+                (implicit hc: HeaderCarrier, ec: ExecutionContext): EitherT[Future, BusinessError, AmountInPence] =
+    for {
+      lt         <- EitherT.liftF(findCurrencyRate(httpClient, calculationRequest.currency, LocalDate.now))
```

Comment:
Related to the last comment - seems a bit wasteful to request the same monthly rates every time?

Don't get me wrong this is what we agreed to do so we can merge on that basis, but I'm wondering whether we need to discuss again?

Reply:
monthly rate? I thought was daily.
Not sure what you mean, are you saying to cash the result or else?